src/dune_rules/pkg_rules.ml

```diff
@@ -259,11 +260,107 @@ module Pkg = struct
     ; info : Pkg_info.t
     ; paths : Paths.t
     ; files_dir : Path.Build.t
+    ; system_provided : bool
```

Comment:
The reason why this flag doesn't make sense is because if a package is "system provided", how can it even have a valid `Pkg.t`? For example, if OCaml is provided by the system, how do you set `paths` correctly?

Reply:
The only reason I added system provided packages was so that their transitive dependencies could be tracked. How about we make this type a variant with a system-provided case and a regular case?

Reply:
If a package is system provided, why do we need to track its (direct or transitive) deps? You would think those deps would system provided as well. Or are they runtime deps of sort?